Objects/TI4_Helpers/TI4_FactionHelper.ttslua

```diff
@@ -1122,6 +1126,703 @@ function tintTokens(params)
+    for _, object in ipairs(getAllObjects()) do
+        if object.tag == 'Bag' and string.match(object.getName(), '^' .. purgeBagName .. '$') then
+            _purgeBagObject = object
+            return _purgeBagObject
+        end
+    end
+
+    assert(false, 'Unable to locate bag with name "' .. purgeBagName .. '"')
+end
+
+local _factionCardNameToAbilityFunc = false
+local _purgeCardQueue = {}
+
+function purgeCard(cardObject)
+    assert(cardObject.tag == 'Card')
+
+    local inHandGuidSet = _zoneHelper.inHand()
+    assert(not inHandGuidSet[cardObject.getGUID()], 'Cannot purge cards that are in a player\'s hand.')
+
+    local purgeBagObject = assert(getPurgeBag())
+
+    --TODO: Maybe deck related stuff? Object deletion checks? Card locked stuff?
+
+    purgeBagObject.putObject(cardObject)
+    broadcastToAll('PURGING card \'' .. cardObject.getName() .. '\'')
+end
+
+local function _heroCardCanBeUsed(cardObject, usingColor)
+    if cardObject.tag ~= 'Card' then
+        print('Cannot use ' .. cardObject.getName() .. '. Hero abilities can only be used from cards.')
+        return false
+    end
+
+    if cardObject.is_face_down then
+        print('Cannot use ' .. cardObject.getName() .. '. Card is facedown; did you unlock it?')
+        return false
+    end
+
+    local inHandGuidSet = _zoneHelper.inHand()
+    if inHandGuidSet[cardObject.getGUID()] then
+        print('Cannot use ' .. cardObject.getName() .. '. Hero cards cannot be used while in your hand.')
+        return false
+    end
+
+    local factionColor = false
+    for color, faction in pairs(allFactions(false)) do
+        if faction.hero == cardObject.getName() then
+            factionColor = color
+            break
+        end
+    end
+    if not factionColor then
+        print('Cannot use ' .. cardObject.getName() .. '. No seated faction has this Hero.')
+        return false
+    end
+
+    if factionColor ~= usingColor and usingColor ~= 'Black' then
+        print('Cannot use ' .. cardObject.getName() .. '. ' .. usingColor .. ' cannot use ' .. factionColor .. '\'s Hero card.')
+        return false
+    end
+
+    -- Any other technical conditions that would prevent us from purging the card or executing the ability?
+    -- Any other gameplay situations that should prevent the Hero card from being used?
+
+    return true
+end
+
+local function _purgeHeroCard(owningObject, clickingColor)
+    if not _heroCardCanBeUsed(owningObject, clickingColor) then
+        -- Function will print why card cannot be used
+        return
+    end
+
+    purgeCard(owningObject)
+end
+
+local function _jacexHeroAbility(owningObject, clickingColor) --Sol Hero
+    if not _heroCardCanBeUsed(owningObject, clickingColor) then
+        -- Function will print why card cannot be used
+        return
+    end
+
+    local factionTokenName = false
+    for _, faction in pairs(allFactions(false)) do
+        if faction.hero == owningObject.getName() then
```

Comment:
This PR is big enough, and I don't know the franken setup tool yet, so I just added another issue for this

Reply:
Making it Franken friendly isn't too bad.  All of the tool-specific Franken handling has been ripped out, replacing it with having tools look at things like faction.abilities instead of the faction name to see if they have an ability.  So basically just need to have franken setup scan for heroes and update the faction.hero entry.  (Can do that in a separate change.)